subprojects/core/src/main/java/org/gradle/initialization/DefaultGradleLauncher.java

```diff
@@ -92,73 +94,17 @@ public SettingsInternal getSettings() {
 
     @Override
     public BuildResult run() {
-        return doBuild(Stage.Build);
+        return buildOperationExecutor.run("Run build", new RunBuildAction(Stage.Build));
```

Comment:
This is the wrong direction to move the operation. Instead of pushing it down further to a common place, we want to move the common place up so that it wraps all invocations, regardless of entry point.

What we should do instead is simply make sure that all `BuildActionRunner` instances are decorated with `RunAsBuildOperationBuildActionRunner`